### DISCUSSION TITLE
Container with lazy line offset

### DISCUSSION THREAD

**Discussion by sisp** [micromark/micromark#127] [Q&A]
Hi there,

I'm trying to create a micromark extension for [Python Markdown admonition](https://python-markdown.github.io/extensions/admonition/) which has a somewhat [peculiar behavior regarding lazy lines](https://github.com/Python-Markdown/markdown/blob/3.4.1/tests/extensions/admonition.txt#L19-L21):

```text
!!! Note "Did you know?"
    The next line is
not a lazy line
```

Unlike standard CommonMark syntax like blockquote where

```
> This is
a lazy line
```

renders as

> This is a lazy line

Python Markdown admonition requires an offset for lazy lines:

```text
!!! Note "Did you know?"
    The next line is
not a lazy line

!!! Note "Did you know?"
    The next line is
 not a lazy line

!!! Note "Did you know?"
    The next line is
  not a lazy line

!!! Note "Did you know?"
    The next line is
   not a lazy line

!!! Note "Did you know?"
    The next line is
    a lazy line

* !!! Note "Did you know?"
      The next line is
not a lazy line

* !!! Note "Did you know?"
      The next line is
  not a lazy line

* !!! Note "Did you know?"
      The next line is
   not a lazy line

* !!! Note "Did you know?"
      The next line is
    a lazy line (somewhat surprisingly)

* !!! Note "Did you know?"
      The next line is
     a lazy line (somewhat surprisingly)

* !!! Note "Did you know?"
      The next line is
      a lazy line
```

Effectively, the logic for determining lazy lines ignores the offset of the fence `!!!` plus at least 4 additional spaces (surprisingly only at least 2 additional spaces when inside a list item).

I've been trying to implement a tokenizer for this kind of behavior using a "document" extension inspired by [blockquote](https://github.com/micromark/micromark/blob/main/packages/micromark-core-commonmark/dev/lib/block-quote.js) and [list](https://github.com/micromark/micromark/blob/main/packages/micromark-core-commonmark/dev/lib/list.js), unfortunately without success regarding lazy lines. For instance,

```text
!!! Note "Did you know?"
    The next line is
not a lazy line
```

is parsed with a lazy line and I can't seem to figure out if and how I can modify the lazy lines logic to ignore the described offset.

I'd very much appreciate some advice. :pray:

**Comment by wooorm** [comment]
Hey! Terribly sorry for the long wait.
Another construct that might be similar are footnotes: https://github.com/micromark/micromark-extension-gfm-footnote.
However, lazy paragraph lines, that’s just a CM thing of paragraphs, and I don’t think you can change it.

It’s been a bit since I worked on `micromark`, I’ll be doing that again the coming months. So I might have a fresher idea about what’s up in the code soon
